model/annotations/task_annotations.go

```diff
@@ -125,6 +137,27 @@ func UpdateAnnotationNote(taskId string, execution int, originalMessage, newMess
 	return errors.Wrapf(err, "updating note for task '%s'", taskId)
 }
 
+// AddMetadataLinkToAnnotation adds a task link to the annotation for the given task.
+func AddMetadataLinkToAnnotation(taskId string, execution int, metadataLink MetadataLink) error {
```

Comment:
nit. If we thread context to the db call below we can capture these in an otel trace.